runtime/src/deterministic.rs

```diff
@@ -836,11 +839,36 @@ impl Clone for Context {
             network: self.network.clone(),
             storage: self.storage.clone(),
             children: self.children.clone(),
+            spawn_config: self.spawn_config,
         }
     }
 }
 
 impl crate::Spawner for Context {
+    fn supervised(&self) -> Self {
+        let mut context = self.clone();
```

Comment:
I think we could avoid this clone (which `with_label` already does)?

Reply:
I'm ok treating `with_label` as a replacement for clone() but treating these as modifiers